Remotion/Data/DomainObjects/Persistence/Rdbms/StorageProviderCommands/IndirectDataContainerLoadCommand.cs

```diff
@@ -23,44 +23,51 @@
   {
-    private readonly IStorageProviderCommand<IEnumerable<ObjectID>, IRdbmsProviderCommandExecutionContext> _objectIDLoadCommand;
-    private readonly IStorageProviderCommandFactory<IRdbmsProviderCommandExecutionContext> _storageProviderCommandFactory;
+    private readonly IRdbmsProviderCommandWithReadOnlySupport<IEnumerable<ObjectID>> _objectIDLoadCommand;
+    private readonly IRdbmsProviderCommandFactory _rdbmsProviderCommandFactory;
 
     public IndirectDataContainerLoadCommand (
-        IStorageProviderCommand<IEnumerable<ObjectID>, IRdbmsProviderCommandExecutionContext> objectIDLoadCommand,
-        IStorageProviderCommandFactory<IRdbmsProviderCommandExecutionContext> storageProviderCommandFactory)
+        IRdbmsProviderCommandWithReadOnlySupport<IEnumerable<ObjectID>> objectIDLoadCommand,
+        IRdbmsProviderCommandFactory rdbmsProviderCommandFactory)
     {
       ArgumentUtility.CheckNotNull("objectIDLoadCommand", objectIDLoadCommand);
-      ArgumentUtility.CheckNotNull("storageProviderCommandFactory", storageProviderCommandFactory);
+      ArgumentUtility.CheckNotNull("rdbmsProviderCommandFactory", rdbmsProviderCommandFactory);
 
       _objectIDLoadCommand = objectIDLoadCommand;
-      _storageProviderCommandFactory = storageProviderCommandFactory;
+      _rdbmsProviderCommandFactory = rdbmsProviderCommandFactory;
     }
 
-    public IStorageProviderCommand<IEnumerable<ObjectID>, IRdbmsProviderCommandExecutionContext> ObjectIDLoadCommand
+    public IRdbmsProviderCommand<IEnumerable<ObjectID>> ObjectIDLoadCommand
     {
       get { return _objectIDLoadCommand; }
     }
 
-    public IStorageProviderCommandFactory<IRdbmsProviderCommandExecutionContext> StorageProviderCommandFactory
+    public IRdbmsProviderCommandFactory RdbmsProviderCommandFactory
     {
-      get { return _storageProviderCommandFactory; }
+      get { return _rdbmsProviderCommandFactory; }
     }
 
-    public IEnumerable<ObjectLookupResult<DataContainer>> Execute (IRdbmsProviderCommandExecutionContext executionContext)
+    public IEnumerable<ObjectLookupResult<DataContainer>> Execute (IRdbmsProviderReadWriteCommandExecutionContext executionContext)
     {
       ArgumentUtility.CheckNotNull("executionContext", executionContext);
 
       var objectIds = _objectIDLoadCommand.Execute(executionContext);
-      return _storageProviderCommandFactory.CreateForSortedMultiIDLookup(objectIds.ToArray()).Execute(executionContext);
+      return _rdbmsProviderCommandFactory.CreateForSortedMultiIDLookup(objectIds.ToArray()).Execute(executionContext);
+    }
+
+    public IEnumerable<ObjectLookupResult<DataContainer>> Execute (IRdbmsProviderReadOnlyCommandExecutionContext executionContext)
+    {
+      ArgumentUtility.CheckNotNull("executionContext", executionContext);
+
+      var objectIds = _objectIDLoadCommand.Execute(executionContext);
+      return _rdbmsProviderCommandFactory.CreateForSortedMultiIDLookup(objectIds.ToArray()).Execute(executionContext);
```

Comment:
```suggestion
      var objectIDs = _objectIDLoadCommand.Execute(executionContext);
      return _rdbmsProviderCommandFactory.CreateForSortedMultiIDLookup(objectIDs.ToArray()).Execute(executionContext);
```

ID is written upper case unless there's precedence, e.g. from the type. It's a historical style mismatch Microsoft started way back when.